src/caf/core/data_structures.py

```diff
@@ -706,5 +1448,292 @@ def old_to_new_dvec(import_data: dict):
+            )
+
+        if self.zoning_system is None:
+            raise ValueError("There is no zoning to remove.")
+
+        # Aggregate all the data
+        summed = fn(self.data, axis=1)
+
+        return DVector(
+            zoning_system=None,
+            segmentation=self.segmentation,
+            time_format=self.time_format,
+            import_data=summed,
+        )
+
+    def sum_zoning(self):
+        """Sum over zones."""
+        return self.remove_zoning()
+
+    def write_sector_reports(
+        self,
+        segment_totals_path: PathLike,
+        ca_sector_path: PathLike,
+        ie_sector_path: PathLike,
+        lad_report_path: PathLike | None = None,
+        lad_report_seg: Segmentation | None = None,
+    ) -> None:
+        """
+        Write segment, CA sector, and IE sector reports to disk.
+
+        Parameters
+        ----------
+        segment_totals_path:
+            Path to write the segment totals report to
+
+        ca_sector_path:
+            Path to write the CA sector report to
+
+        ie_sector_path:
+            Path to write the IE sector report to
+
+        lad_report_path:
+            Path to write the LAD report to
+
+        lad_report_seg:
+            The segmentation to output the LAD report at
+
+        Returns
+        -------
+        None
+        """
+        # Check that not just one argument has been set
+        if bool(lad_report_path) != bool(lad_report_seg):
+            raise ValueError(
+                "Only one of lad_report_path and lad_report_seg has been set. "
+                "Either both values need to be set, or neither."
+            )
+
+        # Segment totals report
+        df = self.sum_zoning().data
+        df.to_csv(segment_totals_path)
+
+        # Segment by CA Sector total reports - 1 to 1, No weighting
+        try:
+            tfn_ca_sectors = ZoningSystem.get_zoning("ca_sector_2020")
+            dvec = self.translate_zoning(tfn_ca_sectors)
+            dvec.data.to_csv(ca_sector_path)
+        except Exception as err:
+            LOG.error("Error creating CA sector report: %s", err)
+
+        # Segment by IE Sector total reports - 1 to 1, No weighting
+        try:
+            ie_sectors = ZoningSystem.get_zoning("ie_sector")
+            dvec = self.translate_zoning(ie_sectors)
+            dvec.data.to_csv(ie_sector_path)
+        except Exception as err:
+            LOG.error("Error creating IE sector report: %s", err)
+
+        if lad_report_seg is None:
+            return
+
+        # Segment by LAD segment total reports - 1 to 1, No weighting
+        try:
+            lad = ZoningSystem.get_zoning("lad_2020")
+            dvec = self.aggregate(lad_report_seg)
+            dvec = dvec.translate_zoning(lad)
+            dvec.data.to_csv(lad_report_path)
+        except Exception as err:
+            LOG.error("Error creating LAD report: %s", err)
+
+    def sum(self):
```

Comment:
Type annotations, I think this returns a float?
```suggestion
    def sum(self) -> float:
```